tracer/src/Datadog.Trace/RuntimeMetrics/RuntimeMetricsWriter.cs

```diff
@@ -129,14 +130,31 @@ internal RuntimeMetricsWriter(IDogStatsd statsd, TimeSpan delay, bool inAzureApp
 
         public void Dispose()
         {
-            AppDomain.CurrentDomain.FirstChanceException -= FirstChanceException;
+            if (_disposed)
+            {
+                return;
+            }
+
+            Log.Debug("Disposing Runtime Metrics");
+            _disposed = true;
             _timer.Dispose();
+            AppDomain.CurrentDomain.FirstChanceException -= FirstChanceException;
+            // We don't dispose runtime metrics on .NET Core because of https://github.com/dotnet/runtime/issues/103480
```

Comment:
Isn't this fixed in .NET 9? Should we add a runtime check (since we can't use `#if NET_9_0_OR_GREATER`) and dispose if .NET version is >= 9?